elastic/kibana
# [Cloud Security] Adding Auth check on API

## Summary

This PR addresses the issue where currently only some of our API does Auth check. All of our API should be doing this. Furthermore we are adding new API FTR to cover this scenario as well

Notes:
Currently Benchmark related API FTR is blocked by https://github.com/elastic/kibana/issues/188059

How to test:

1. Create a user role with read access to Security and All access for Integration, Fleet and Saved Data management. Then assign this role to a new user
2. Create a user role with All access for Integration, Fleet and Saved Data management. Then assign this role to a new user.
3. Try calling the API as user created on step 1 and they should be able to receive non 403 status code
4. Try calling the API as user created on step 2 and they should receive  403 status code
